__tests__/app.test.js

```diff
@@ -284,3 +278,110 @@ describe("POST /api/articles/article_id/comments", () => {
+        })
+    });
+    test("Should respond with 404 and send an error message if valid id but does not exist in database", () => {
+        return request(app).patch("/api/articles/123").send({
+            incVotes: 10
+        }).expect(404).then((response) => {
+            const text = response.error.text;
+            expect(text).toEqual("{\"status\":404,\"error\":\"Not Found\",\"message\":\"Not found\"}");
+        });
+    });
+    test("Should respond with 500 and send an error message in case of an error", () => {
+        jest.spyOn(models, 'fetchPatchArticleByArticleId').mockRejectedValue(new Error("Database query error"));
+        return request(app).patch("/api/articles/1").send({
+            incVotes: 10
+        }).expect(500).then((response) => {
+            const text = response.error.text;
+            expect(text).toEqual("{\"status\":500,\"error\":\"Internal Server Error\"}");
+        });
+    });
+    test("Should return 400 and send an error message if request body is undefined", () => {
+        return request(app).patch("/api/articles/1").send({}).expect(400).then((response) => {
+            const text = response.error.text;
+            expect(text).toBe("{\"status\":400,\"error\":\"Bad Request\",\"message\":\"Invalid request body\"}")
+        })
+    });
+    test("Should return 400 and send an error message if request body is not a number", () => {
+        return request(app).patch("/api/articles/1").send({ incVotes: "Hello" }).expect(400).then((response) => {
+            const text = response.error.text;
+            expect(text).toBe("{\"status\":400,\"error\":\"Bad Request\",\"message\":\"Invalid request body\"}")
+        })
+    });
+})
+
+describe("DELETE /api/comments/:comment_id", () => {
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+    test("Should respond with 204 and no response", async () => {
+        return request(app).delete("/api/comments/1").expect(204);
+      });
+    test("Should respond with 400 and error message if comment id is invalid", () => {
+        return request(app).delete("/api/comments/0").expect(400).then((response) => {
+            const text = response.error.text;
+            expect(text).toBe("{\"status\":400,\"error\":\"Bad Request\",\"message\":\"Invalid query\"}")
+        })
+    });
+    test("Should respond with 404 and error message if id is valid but not in database", () => {
+        return request(app).delete("/api/comments/123").expect(404).then((response) => {
+            const text = response.error.text;
+            expect(text).toBe("{\"status\":404,\"error\":\"Not Found\",\"message\":\"Not found\"}")
+        })
+    });
+    test("Should respond with 500 and send an error message in case of an error", () => {
+        jest.spyOn(models, 'fetchDeleteCommentsByCommentId').mockRejectedValue(new Error("Database query error"));
+        return request(app).delete("/api/comments/1").expect(500).then((response) => {
```

Comment:
If testing for a 500 you would only need to do this once and is more of a generic test rather than for a specific endpoint.